src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+ m_current_iteration = 0;
+ m_complete = false;
+ register_params();
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+}
+
+bool CIterativeLinearMachine::train(CDotFeatures* f, CLabels* l)
+{
+   init_model(f, l);
+   continue_train(f, l);
+   return m_complete;
+}
+
+void CIterativeLinearMachine::continue_train(CDotFeatures* f, CLabels* l)
+{
+  //resume_computation();
+   while (m_current_iteration < m_max_iterations && !m_complete)
+   {
+       COMPUTATION_CONTROLLERS
+       iteration(f, l);
+       m_current_iteration++;
+   }
+}
```

Comment:
You could call here, right before the end of ``continue_train``, the method  `reset_computation_variables` (see [here](https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/lib/StoppableSGObject.h#L85)). So that each time the `continue_train` method finishes, you can call it again.

Reply:
yeah that works